evalml/tests/pipeline_tests/test_pipeline_utils.py

```diff
@@ -40,465 +40,204 @@
-                "text for a column, this should be a text column!!",
-                "text string",
-                "hello world",
-            ],
-        }
-    )
-    y = pd.Series([0, 0, 1, 1, 0])
-    if input_type == "ww":
-        X.ww.init()
-        y = ww.init_series(y)
-    estimators = get_estimators(problem_type=problem_type)
-
-    pipeline_class = _get_pipeline_base_class(problem_type)
-    if problem_type == ProblemTypes.MULTICLASS:
-        y = pd.Series([0, 2, 1, 2])
-
-    for estimator_class in estimators:
-        if problem_type in estimator_class.supported_problem_types:
-            parameters = {}
-            if is_time_series(problem_type):
-                parameters = {
-                    "pipeline": {"date_index": None, "gap": 1, "max_delay": 1},
-                    "Time Series Baseline Estimator": {
-                        "date_index": None,
-                        "gap": 1,
-                        "max_delay": 1,
-                    },
-                }
-
-            pipeline = make_pipeline(X, y, estimator_class, problem_type, parameters)
-            assert isinstance(pipeline, pipeline_class)
-            delayed_features = []
-            if is_time_series(problem_type):
-                delayed_features = [DelayedFeatureTransformer]
-            if estimator_class.model_family == ModelFamily.LINEAR_MODEL:
-                estimator_components = [OneHotEncoder, StandardScaler, estimator_class]
-            elif estimator_class.model_family == ModelFamily.CATBOOST:
-                estimator_components = [estimator_class]
-            else:
-                estimator_components = [OneHotEncoder, estimator_class]
-            if estimator_class.model_family == ModelFamily.ARIMA:
-                expected_components = [TextFeaturizer, Imputer] + estimator_components
-            else:
-                expected_components = (
-                    [TextFeaturizer, Imputer] + delayed_features + estimator_components
-                )
-            pipeline.component_graph.compute_order == [
-                component.name for component in expected_components
-            ]
-
+@pytest.mark.parametrize(
+    "test_description, column_names",
+    [
+        ("all nan is not categorical", ["all_null", "numerical"]),
+        ("all types", ["all_null", "categorical", "dates", "numerical"]),
+        ("no all_null columns", ["numerical", "categorical", "dates"]),
+        ("date, numerical", ["dates", "numerical"]),
+        ("only text", ["text"]),
+        ("only dates", ["dates"]),
+        ("only numerical", ["numerical"]),
+        ("only all_null", ["all_null"]),
+        ("only categorical", ["categorical"]),
+        ("text with other features", ["text", "numerical", "categorical"]),
+        ("url with other features", ["url", "numerical", "categorical"]),
+        ("email with other features", ["email", "numerical", "categorical"]),
+    ],
+)
+def test_make_pipeline_master(
```

Comment:
`test_description` isn't being used by the code but since we no longer have the title of tests to tell us what is going on, I thought this would be useful to quickly understand what each test case is for.

Reply:
Can we use it in the assert error message?